frontend/src/metabase/lib/visualization_settings.js

```diff
@@ -1,230 +1,573 @@
+        });
+    }
 }
 
-function getSettingsForGroup(dbSettings, groupName) {
-    if (typeof dbSettings != "object") {
-        dbSettings = {};
+function getDefaultDimensionsAndMetrics([{ data: { cols, rows } }]) {
+    let type = getChartTypeFromData(cols, rows, false);
+    switch (type) {
+        case DIMENSION_DIMENSION_METRIC:
+            let dimensions = [cols[0], cols[1]];
+            if (isDate(dimensions[1]) && !isDate(dimensions[0])) {
+                // if the series dimension is a date but the axis dimension is not then swap them
+                dimensions.reverse();
+            } else if (dimensions[1].cardinality > dimensions[0].cardinality) {
+                // if the series dimension is higher cardinality than the axis dimension then swap them
+                dimensions.reverse();
+            }
+            return {
+                dimensions: dimensions.map(col => col.name),
+                metrics: [cols[2].name]
+            };
+        case DIMENSION_METRIC:
+            return {
+                dimensions: [cols[0].name],
+                metrics: [cols[1].name]
+            };
+        case DIMENSION_METRIC_METRIC:
+            return {
+                dimensions: [cols[0].name],
+                metrics: cols.slice(1).map(col => col.name)
+            };
+        default:
+            return {
+                dimensions: [null],
+                metrics: [null]
+            };
     }
+}
 
-    if (typeof settings[groupName] == "undefined") {
-        return dbSettings;
+function getDefaultDimensionAndMetric([{ data: { cols, rows } }]) {
+    const type = getChartTypeFromData(cols, rows, false);
+    if (type === DIMENSION_METRIC) {
+        return {
+            dimension: cols[0].name,
+            metric: cols[1].name
+        };
+    } else {
+        return {
+            dimension: null,
+            metric: null
+        };
     }
+}
+
+function getOptionFromColumn(col) {
+    return {
+        name: getFriendlyName(col),
+        value: col.name
+    };
+}
 
-    if (typeof dbSettings[groupName] == "undefined") {
-        dbSettings[groupName] = {};
+const CURRENCIES = ["afn", "ars", "awg", "aud", "azn", "bsd", "bbd", "byr", "bzd", "bmd", "bob", "bam", "bwp", "bgn", "brl", "bnd", "khr", "cad", "kyd", "clp", "cny", "cop", "crc", "hrk", "cup", "czk", "dkk", "dop", "xcd", "egp", "svc", "eek", "eur", "fkp", "fjd", "ghc", "gip", "gtq", "ggp", "gyd", "hnl", "hkd", "huf", "isk", "inr", "idr", "irr", "imp", "ils", "jmd", "jpy", "jep", "kes", "kzt", "kpw", "krw", "kgs", "lak", "lvl", "lbp", "lrd", "ltl", "mkd", "myr", "mur", "mxn", "mnt", "mzn", "nad", "npr", "ang", "nzd", "nio", "ngn", "nok", "omr", "pkr", "pab", "pyg", "pen", "php", "pln", "qar", "ron", "rub", "shp", "sar", "rsd", "scr", "sgd", "sbd", "sos", "zar", "lkr", "sek", "chf", "srd", "syp", "tzs", "twd", "thb", "ttd", "try", "trl", "tvd", "ugx", "uah", "gbp", "usd", "uyu", "uzs", "vef", "vnd", "yer", "zwd"];
```

Comment:
not a big deal but this seems like the kind of thing that might be best served as an import from a util somewhere.